frontend/src/scenes/performance/WebPerformance.tsx

```diff
@@ -85,6 +85,24 @@ const overlayFor = (resourceTiming: ResourceTiming): JSX.Element => {
     )
 }
 
+const MouseTriggeredPopUp = ({
+    content,
+    children,
+}: {
+    content: JSX.Element
+    children: React.ReactNode
+}): JSX.Element => {
+    const [mouseIsOver, setMouseIsOver] = useState(false)
+
+    return (
+        <Popup overlay={content} visible={mouseIsOver} maxWidth={'300px'}>
+            <div onMouseEnter={() => setMouseIsOver(true)} onMouseLeave={() => setMouseIsOver(false)}>
+                {children}
+            </div>
+        </Popup>
+    )
+}
```

Comment:
This should ideally be abstracted away, ideally into `Popup`, so that you can just set `<Popup trigger="hover">` (or, by default, `trigger="click"`) and not worry about `onMouseEnter` etc. I feel like `Popup` needs a bit of a refactor overall to 1. be named more accurately (in UX this kind of thing is called a popover, a popup is something else) 2. abstract away the logic of closing/opening for most cases – that's probably outside of the scope here though.

Reply:
> This should ideally be abstracted away, ideally into Popup

Yep, agreed. Was surprised I had to make it... But was also trying to keep scope here smaller :)